### ISSUE TITLE
Global search parallelism is gone. RIP.

### ISSUE THREAD

**Issue by makemefeelgr8** [tachiyomiorg/tachiyomi#8906]
### Steps to reproduce

...open the source code and review it? There's no proper template for reporting finds like this one.

### Expected behavior

-

### Actual behavior

-

### Crash logs

-

### Tachiyomi version

upstream

### Android version

13

### Device

-

### Other details

Well, here we go again.
[The PR to blame.](https://github.com/tachiyomiorg/tachiyomi/pull/8631)
Coroutine dispatcher has has 5 threads [here](https://github.com/tachiyomiorg/tachiyomi/blob/master/app/src/main/java/eu/kanade/tachiyomi/ui/browse/source/globalsearch/SearchScreenModel.kt#L39). It's kind of a weird number- ever seen a CPU with 5 threads? I'd rather stick to defaults, which makes it equal the amount of CPU threads, but whatever.
Here's the code:

    private val coroutineDispatcher = Executors.newFixedThreadPool(5).asCoroutineDispatcher()

And [this](https://github.com/tachiyomiorg/tachiyomi/blob/master/app/src/main/java/eu/kanade/tachiyomi/ui/browse/source/globalsearch/SearchScreenModel.kt#L135) is how the search works.
Behold, more code:

        coroutineScope.launch {
            sources.forEach { source ->
                val page = try {
                    withContext(coroutineDispatcher) {
                        source.fetchSearchManga(1, query, source.getFilterList()).awaitSingle()
                    }
                }
                ...
            }

Am I the only one who can see the issue here? It's not like "run the coroutine for each source" to search them in parallel. It's like "run the coroutine, but search all sources in a synchronous forEach loop one by one". Which kind of defeats the whole purpose of specifying `5` threads in the coroutine definition. Cause the new-aka-Compose search looks pretty synchronous to me. Or I'm the one being stupid here, which might also be the case.

Anyway, this code definitely needs some love. Quite a lot of it, actually.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.14.3](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.
